Add honorifics and validator

gbirke opened the pull request:
Localizable honorifics can now be passed to the presentation layer and
validated by the validators.

Moved the German locale to the default configuration.

JeroenDeDauw commented:
Needs rebase.

Gabriel and I talked about this and figured it's nicer to use a value object

gbirke commented:
Rebased, using new `ValidationResult` class.
Simplified Honorifics class as proposed by @JeroenDeDauw: It now is a very simple ValueObject, the locale-specific stuff is done in FunFunFactory

JeroenDeDauw commented:
+1
